src/a_star_search/core.clj

```diff
@@ -1,27 +1,27 @@
 (ns a-star-search.core
   (:require [search.algorithms :as alg]
-            [problems.n-puzzle :as np])
+            [problems.chinese-ring-puzzle :as crp])
   (:gen-class))
 
 (defn print-results [result path costs goal-state]
   (println (str "We explored " (count result) " states."))
   (println "The path to the solution is:")
-  (doseq [b (map :board path)]
+  (doseq [b (map :Rings path)]
     (println b))
   (when costs
     (println "The path has" (count path) "steps and its cost is" (costs goal-state))))
 
 (defn -main
   "Search for a hard-coded N-puzzle target state."
   [& args]
-  (let [start-state (np/->State [[0 1 3] [4 2 5] [7 8 6]] [0 0])
-        goal-state (np/->State [[0 1 2] [3 4 5] [6 7 8]] [0 0])
-        max-states 1000000
+  (let [start-state (crp/->State 0 ["on" "on" "on" "on" "on" "on" "on" "on" "on"])
```

Comment:
It probably would have been nice to have pulled `start-state` and `goal-state` out of `-main` and into their own `def`s so they don't clutter `-main` up so much, especially since they're long.

Also, it would have been more "Clojure-y" to have used keywords (e.g., `:on` and `:off`) rather than strings for the specific ring states.